### ISSUE TITLE
Task: Use block patterns API instead of hard-coding patterns

### ISSUE THREAD

**Issue by alshakero** [Automattic/wp-calypso#67195]
### Details

Currently, we have hard-coded patterns listed [here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/pattern-picker/src/patterns.ts).

This is blocked by the patterns API not being publicly accessible, but a12ns aren't affected by this, so let's start the work but not merge until it is (p1661933400326759/1661873432.122819-slack-C02M88KJ684). 

These will be a constant maintenance burden, we should use this API to fetch fresh patterns. The API's URL is `https://public-api.wordpress.com/wp/v2/sites/210002353/block-patterns/patterns?http_envelope=1`. We need to fetch this pattern in a react-query hook, and render the ones that start with `lynx` in the pattern-picker [here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/pattern-picker/src/index.tsx#L60).

This will also need some changes to the store. When the user picks a pattern, we now store the `patternId: number`, we should move away from that and store an object with two values `{ name: string, content: string }`. See one of the hard-coded patterns to see what these props represent.

And then, when we apply the pattern [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/hooks/use-setup-onboarding-site.ts#L100-L110), we should get the stored value and upload it. 

### Checklist

- [ ] Make sure no pattern is left hard-coded.
- [ ] Make sure the previews work as expected. 
- [ ] Verify that the patterns are applied correctly.

**Comment by alshakero** [comment]
Closing as won't fix.
